src/protocols/secure_channel/CASESession.cpp

```diff
@@ -282,6 +265,8 @@ CHIP_ERROR CASESession::EstablishSession(const Transport::PeerAddress peerAddres
     mExchangeCtxt->SetResponseTimeout(kSigma_Response_Timeout);
     mConnectionState.SetPeerAddress(peerAddress);
     mConnectionState.SetPeerNodeId(peerNodeId);
+    // TODO: implement API for this in OperationalCredentialSet
+    mTrustedRootId = operationalCredentialSet->GetOpCredSet()[opCredSetIndex].GetCertSet()[0].mAuthKeyId;
```

Comment:
This indexing is unsafe. There must be bound checks to ensure the indices are correct. Perhaps step 1 is to implement the API needed in OperationalCredentialSet, and then be able to rely on it.

Reply:
Updated.